src/coap_gnutls.c

```diff
@@ -828,6 +832,22 @@ post_client_hello_gnutls(gnutls_session_t g_session)
   return ret;
 }
 
+/*
+ * return 0   Success (GNUTLS_E_SUCCESS)
+ *        neg GNUTLS_E_* error code
+ */
+static int
+post_client_hello_gnutls_psk(gnutls_session_t g_session)
+{
+  coap_session_t *c_session =
+                (coap_session_t *)gnutls_transport_get_ptr(g_session);
+  coap_gnutls_env_t *g_env = (coap_gnutls_env_t *)c_session->tls;
+  int ret = GNUTLS_E_SUCCESS;
+
+  g_env->seen_client_hello = 1;
+  return ret;
```

Comment:
For clarity, I suggest removing the local variable and write `return GNUTLS_E_SUCCESS;` instead.

Reply:
Updated